### PR TITLE
feat: add modulo function

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Implement the `mod` function to find remainders

 - added a unit test
 - updated the sqlness test

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/3140

### CODE REVIEW

File: src/common/function/src/scalars/math/modulo.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+impl Function for ModuloFunction {
+    fn name(&self) -> &str {
+        NAME
+    }
+
+    fn return_type(
+        &self,
+        _input_types: &[ConcreteDataType],
+    ) -> common_query::error::Result<ConcreteDataType> {
+        Ok(ConcreteDataType::int64_datatype())
```
Comment (waynexia) [inline src/common/function/src/scalars/math/modulo.rs:52]: What if the input is a float number

File: src/common/function/src/scalars/math/modulo.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::uniform(
+            2,
+            vec![
+                ConcreteDataType::int8_datatype(),
+                ConcreteDataType::int16_datatype(),
+                ConcreteDataType::int32_datatype(),
+                ConcreteDataType::int64_datatype(),
```
Comment (waynexia) [inline src/common/function/src/scalars/math/modulo.rs:62]: It seems like `uint8` to `uint64` can also be supported

**Review by waynexia — APPROVED** [review]
Mostly looks good to me, thanks @etolbakov ❤️

File: src/common/function/src/scalars/math/modulo.rs [hunk 3]
```diff
@@ -0,0 +1,241 @@
+// Copyright 2023 Greptime Team
```
Comment (etolbakov) [inline src/common/function/src/scalars/math/modulo.rs:1]: Not sure if that should be updated @waynexia

**Comment by etolbakov** [comment]
@killme2008 @waynexia could you please take a look when you have time?

**Review by killme2008 — APPROVED** [review]
LGTM! Thank u!

File: src/common/function/src/scalars/math/modulo.rs (hunk 3 above)
Comment (killme2008) [reply src/common/function/src/scalars/math/modulo.rs:1]: It's all right to not update it.

File: src/common/function/src/scalars/math/modulo.rs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        let nums = vec![18, -17, 5, -6];
+        let divs = vec![4, 8, -5, -5];
+
+        let args: Vec<VectorRef> = vec![
+            Arc::new(Int32Vector::from_vec(nums.clone())),
+            Arc::new(Int32Vector::from_vec(divs.clone())),
+        ];
+        let result = function.eval(FunctionContext::default(), &args).unwrap();
+        assert_eq!(result.len(), 4);
+        for i in 0..3 {
```
Comment (dimbtp) [inline src/common/function/src/scalars/math/modulo.rs:131]: Is this `0..3` should be `0..4`?

Comment (waynexia) [reply src/common/function/src/scalars/math/modulo.rs:131]: Good catch! I haven't noticed this before 👀 

Are you willing to submit a fix?

Comment (etolbakov) [reply src/common/function/src/scalars/math/modulo.rs:131]: Indeed 🙈!
Thanks @dimbtp!
Feel free to fix it or I can do it in the couple of hours

**Pull request merged**